core/spec/models/spree/address_spec.rb

```diff
@@ -119,6 +119,47 @@
     end
   end
 
+  context "when updating a record" do
+    context "if use_combined_first_and_last_name_in_address is set to false (default)" do
+      before do
+        allow(Spree::Config).to receive(:use_combined_first_and_last_name_in_address) { false }
+      end
+
+      context "and the `name` attribute is not in changeset" do
+        let(:address) { create(:address, firstname: 'John', lastname: 'Doe') }
+
+        it "sets `name` from `firstname` and `lastname`" do
+          new_address = described_class.immutable_merge(address, firstname: 'Jane', lastname: 'Fonda')
+          expect(new_address.read_attribute(:name)).to eq('Jane Fonda')
+        end
+      end
```

Comment:
Hey @tvdeyen, did you intentionally leave out the `"and the `name` attribute is in changeset"` context here? Isn't that exactly the scenario we want to fix?

Reply:
No, we want to test exactly that scenario. Given are only `firstname` and `lastname` in the params. See the issue https://github.com/solidusio/solidus/issues/4094

Reply:
Ah got it, so essentially `name` is there because it is part of the address already (being "in sync" with firstname/lastname).

What do you think about adding the other context as well (passing the `name` explicitly with `use_combined_first_and_last_name_in_address: false`)? I understand that this should never happen but at least we have a spec that describe the state of the system in that case.

Reply:
Done